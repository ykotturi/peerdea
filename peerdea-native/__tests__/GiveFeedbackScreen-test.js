import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow'

import GiveFeedback from '../screens/GiveFeedbackScreen';

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);
jest.mock('react-native-autolink', () => {
  return {
    Autolink: jest
      .fn()
      .mockImplementation(({ children }) => children),
  };
});

describe('Give Feedback Screen', () => {
  jest.useFakeTimers();

  const mockNavigation = {
    navigate: jest.fn(),
    addListener: jest.fn().mockImplementation((event, callback) => {
      callback();
      //returning value for `navigationSubscription`
      return { remove: jest.fn() }
    }),
  };
  const renderer = new ShallowRenderer();

  it(`renders the give feedback screen loading`, () => {
    const tree = renderer.render(<GiveFeedback navigation={mockNavigation} />);
    expect(tree).toMatchSnapshot();
  });
});
