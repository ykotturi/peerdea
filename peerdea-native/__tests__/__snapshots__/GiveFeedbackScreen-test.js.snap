// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Give Feedback Screen renders the give feedback screen loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 37.5,
        "paddingVertical": 5,
      }
    }
  >
    Loading concepts...
  </Text>
</View>
`;
